portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +27,108 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+//*********************** LOADING SCREEN ***********************
+function setVisible(selector, bool) {
+    const el = document.getElementById(selector);
+    el.classList.toggle('hiddenclass', bool);
+}
+
+function swapLoaderForContent(){
+    setVisible('header', false);
+    setVisible('content', false);
+    setVisible('loaderid', true);
+}
+
+//*********************** SEARCH BAR FUNCTIONS ***********************
+/**
+  * Swaps the loader for the content and enables search
+  */
+window.onload = function(){
+    swapLoaderForContent();
+    document.getElementById('searchbar').onkeydown = function(e){
+        if(e.keyCode == 13){
+            runSearch(document.getElementById('searchbar')
+                .value.toLocaleLowerCase());
+        }
+    };
+}
+
+
+
+//*********************** SEARCH IMPLEMENTATION ***********************
+/**
+  * Retreives the HTML from the source
+  */
+function retrieveHtml(){
+    var source = document.body.innerHTML;
+
+    if (source != null){
+        return source.toLocaleLowerCase();    //toLocaleLowerCase() to ignore caps
+    } else {
+        alert("source is null");
+    }
+}
+
+
+/**
+  * Finds the closest tag before the provided index in the html text.
+  */
+function getClosestTag(index, source){
+    while (source.charAt(index) != '>'){
+        if (index == 0){
+            alert("index hit 0");
+            break;
+        }
+        index--;
+    }
+
+    var endIndex = index+1; //plus 1 to account for bracket
+
+    while (source.charAt(index) != '<'){
+        index--;
+    }
+
+    var startIndex = index;
+
+    return source.substring(startIndex, endIndex)
+}
+
+/**
+  * Gets HTML id from given word
+  */
+function retrieveId(string, source){
+    var searchForId = string.search("id=\"");
+    if (!searchForId){
+        return;
+    }
+    const startIndex = searchForId+4;
+    var retString = string.charAt(startIndex);
+
+    for(var i = startIndex+1; string.charAt(i) != "\""; i++){
+        retString += string.charAt(i);
+    }
+
+    return retString;
+}
+
+/**
+  * Runs the search
```

Comment:
Consider writing a comment that helps someone new to this code understand this function better. What is the input it takes? What is it searching for/on?

This has some helpful info on writing function comments: https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments